### PR TITLE
Add Dirac-delta support

### PR DESCRIPTION
**Pull request by brandonwillard**
This PR adds Dirac-delta support via an ephemeral `DiracDelta` `Op`.  The relevant lifting and removal rewrites have been added as well.

### CODE REVIEW

File: aeppl/logprob.py [hunk 1]
```diff
@@ -596,3 +596,10 @@ def multinomial_logprob(op, values, *inputs, **kwargs):
         at.all(at.ge(n, 0)),
     )
     return res
+
+
+@_logprob.register(DiracDelta)
+def diracdelta_logprob(op, values, *inputs, **kwargs):
+    (values,) = values
+    (const_value,) = inputs
+    return at.switch(at.eq(values, const_value), 0.0, -np.inf)
```
Comment (ricardoV94) [inline aeppl/logprob.py:605]: Os strict equality or `isclose` best here?

**Review by ricardoV94 — COMMENTED** [review]
Why not make it a RandomVariable? thay way the default transforms would work out of the box, and we can still have the remove useless rewrite.

**Comment by brandonwillard** [comment]
> Why not make it a RandomVariable? thay way the default transforms would work out of the box, and we can still have the remove useless rewrite.

`RandomVariable` interface simply doesn't fit this type of "distribution" well enough, so there would be a bit more boilerplate to that.  Plus, it would only afford a minor convenience as a result.  Also, this type of `Op` has more simplifying properties that are easy to express directly, whereas `RandomVariable` doesn't, so this `Op` would be unnecessarily restricted by what was available to `RandomVariable`s&mdash;unless we implemented its special properties (e.g. nearly every `Op` can be lifted through it), but then we're already doing more work than the current approach.

File: aeppl/logprob.py (hunk 1 above)
Comment (brandonwillard) [reply aeppl/logprob.py:605]: Yeah, we could use that.

**Comment by ricardoV94** [comment]
Yeah nevermind. I was thinking only about the Dimshuffle and broadcast lifts, but here you are doing it for every unary operator as well.

**Review by ricardoV94 — DISMISSED** [review]

**Pull request merged**